### PR TITLE
Update golang version

### PR DESCRIPTION
**Pull request by sanfern**
Bump up go version to 1.18.6
Build on ubuntu 20.04
Building static binary
  
Signed-off-by: Santhosh Fernandes <santhosh.fernandes@gmail.com>

### CODE REVIEW

**Review by jniesz — DISMISSED** [review]
LGTM

File: kf/bpfmap.go [hunk 1]
```diff
@@ -33,23 +33,24 @@ type MetricsBPFMap struct {
 	lastValue  float64
 }
 
-// This function is used to update eBPF maps, which are used by network functions.
+// Update function is used to update eBPF maps, which are used by network functions.
```
Comment (jniesz) [inline kf/bpfmap.go:36]:
```suggestion
// The update function is used to update eBPF maps, which are used by network functions.
```

**Review by jniesz — DISMISSED** [review]

**Review by dthaler — DISMISSED** [review]

**Review by dthaler — DISMISSED** [review]

File: .github/workflows/ci-build.yaml [hunk 2]
```diff
@@ -46,9 +45,9 @@ jobs:
         run: |
           go test ./...
 
-      - uses: dominikh/staticcheck-action@4ec9a0dff54be2642bc76581598ba433fd8d4967
+      - uses: dominikh/staticcheck-action@v1.2.0
```
Comment (jniesz) [inline .github/workflows/ci-build.yaml:48]: Shouldn't we use hash here?

File: Makefile [hunk 3]
```diff
@@ -9,7 +9,7 @@ swagger:
 	@$(GOPATH)/bin/swag init -d "./" -g "apis/configwatch.go"
 
 build:
-	@go build
+	@CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="-extldflags=-static"
```
Comment (jniesz) [inline Makefile:12]: I think we can drop the GOOS and GOARCH lines as it should pick up the correct defaults.  I would also drop 'ldflags'.  We should only need 'CGO_ENABLED=0 go build' to build static binaries or 'go build -tags osusergo,netgo'

**Review by jniesz — APPROVED** [review]
LGTM

**Review by charleskbliu0 — APPROVED** [review]
Looks good.

**Review by jaysheth2 — APPROVED** [review]

**Pull request merged**